### PR TITLE
2021 03 18 ralucarbc javascript some more methods and functions

### LINKED ISSUE #47: feat: sum mixed array
* [x] create the functionality
* [x] display it in the browser

### LINKED ISSUE #48: feat: all unique
* [x] create the functionality
* [x] display it in the browser

### LINKED ISSUE #49: feat: squares sequence
* [x] create the functionality
* [x] display it in the browser

### PR DESCRIPTION
**Pull request by RalucaRbc**
this closes #47 
this closes #48 
this closes #49

### CODE REVIEW

**Review by andreimaff — APPROVED** [review]
niceee 💯

**Review by Constantin-Matei — APPROVED** [review]
Nice!✔

**Review by M00N89 — APPROVED** [review]
stupendo 💯

File: web/ralucarbc/javascript/all-unique/all-unique.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+const divOutput = document.querySelector("#output");
+const divFunction = document.querySelector("#function");
+
+h2Instructions.textContent = `Write a program to determine if a string contains only unique characters. Return true if it does and false otherwise. The string may contain any of the 128 ASCII characters. Characters are case-sensitive, e.g. 'a' and 'A' are considered different characters.`;
+
+let string = "++-";
+divContent.textContent = `The test string is: "` + string + `".`;
+
+const hasUniqueChars = (str) =>
+  str.length === [...new Set(str.split(""))].length ? true : false;
```
Comment (QuintyHH) [inline web/ralucarbc/javascript/all-unique/all-unique.js]: By default returns boolean, no need to use ternary (you can safely remove ? true : false, the outcome doesn't change)

File: web/ralucarbc/javascript/squares-sequence/squares-sequence.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  if (n <= 0) return [];
+  else {
+    let arr = new Array(n);
+    arr[0] = x;
+    for (i = 1; i < arr.length; i++) {
+      arr[i] = Math.pow(arr[i - 1], 2);
+    }
+    return arr;
+  }
+}
```
Comment (QuintyHH) [inline web/ralucarbc/javascript/squares-sequence/squares-sequence.js]: This function could use a refactor, I suggest you use arrow functions, and replace the 'for' loop with a map. Also, declaring the array as a new instance of the Array class is VERY 1985.

Good use of guard clause on 18, but there is no need for an 'else' statement on 19, as that defeats the purpose of the early return (The whole idea is to not increase indentation).

File: web/ralucarbc/javascript/squares-sequence/squares-sequence.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+const h2Instructions = document.querySelector("#instructions");
+const divContent = document.querySelector("#content");
+const divOutput = document.querySelector("#output");
+const divFunction = document.querySelector("#function");
+
+h2Instructions.textContent = `Complete the function that returns an array of length n, starting with the given number x and the squares of the previous number. If n is negative or zero, return an empty array/list.`;
```
Comment (QuintyHH) [inline web/ralucarbc/javascript/squares-sequence/squares-sequence.js:6]: I've seen this throughout your commits. The use of ticks (Template String) and concatenation as well. 
If ES6 is permitted, feel free to use just Template String (Ex: \`This is my string bit to which I add my ${variable}\`)
Else, use old school string concat (Ex: "This is my string to which I add my " + variable)

**Review by QuintyHH — CHANGES_REQUESTED** [review]
Sorry I have to do this to you, Raluca :)

**Review by marianzburlea — CHANGES_REQUESTED** [review]
for -> map
let's have a chat

**Review by marianzburlea — APPROVED** [review]
💯

File: web/ralucarbc/javascript/all-unique/all-unique.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+const divOutput = document.querySelector("#output");
+const divFunction = document.querySelector("#function");
+
+h2Instructions.textContent = `Write a program to determine if a string contains only unique characters. Return true if it does and false otherwise. The string may contain any of the 128 ASCII characters. Characters are case-sensitive, e.g. 'a' and 'A' are considered different characters.`;
+
+let string = "++-";
+divContent.textContent = `The test string is: "${string}".`;
+
+const hasUniqueChars = (str) =>
+  str.length === [...new Set(str.split(""))].length;
```
Comment (QuintyHH) [inline web/ralucarbc/javascript/all-unique/all-unique.js]: no need to split, [...new Set(str)].length is enough

**Review by QuintyHH — CHANGES_REQUESTED** [review]
Final changes plix

**Review by QuintyHH — APPROVED** [review]

**Pull request merged**